src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -456,21 +456,17 @@ void Terminal::_WriteBuffer(const std::wstring_view& stringView)
         {
             // TODO: MSFT 21006766
             // This is not great but I need it demoable. Fix by making a buffer stream writer.
-            if (wch >= 0xD800 && wch <= 0xDFFF)
-            {
-                const OutputCellIterator it{ stringView.substr(i, 2), _buffer->GetCurrentAttributes() };
-                const auto end = _buffer->Write(it);
-                const auto cellDistance = end.GetCellDistance(it);
-                i += cellDistance - 1;
-                proposedCursorPosition.X += gsl::narrow<SHORT>(cellDistance);
-            }
-            else
-            {
-                const OutputCellIterator it{ stringView.substr(i, 1), _buffer->GetCurrentAttributes() };
-                const auto end = _buffer->Write(it);
-                const auto cellDistance = end.GetCellDistance(it);
-                proposedCursorPosition.X += gsl::narrow<SHORT>(cellDistance);
-            }
+            //
+            // If wch is a surrogate character we need to read 2 code units
+            // from the stringView to form a single code point.
+            const auto isSurrogate = wch >= 0xD800 && wch <= 0xDFFF;
+            const size_t codePointLength = isSurrogate ? 2 : 1;

```

Comment:
Isn't `codePointLength` always going to be the same value as `GetInputDistance` between the two iterators? Maybe I chose wrong with `GetCellDistance`, but it feels like it was just `GetInputDistance` instead.

Reply:
Yes we could use `GetInputDistance()` instead. I checked & it works as well.
Do you want me to use ``GetInputDistance()` instead?

Reply:
Yeah just because GetCellDistance and GetInputDistance are meaningful sentinels to me of where all in the code someone is doing this type of calculation. (a.k.a. places where this sort of thing goes to hell all the time).

Reply:
I'll change it. The reason I _personally_ disliked using `GetInputDistance()` until now is because I expect the `TextBuffer::Write` function to consume all given code units in the `OutputCellIterator` and advance `i` accordingly.
This is only the case if you use the same variable you use for `stringView.substr()`, as for incrementing `i`. Again, I'll change it though, because you make a good point.

Reply:
As a sentinel, if we decide that `OutputCellIterator` is globally stupid for making the caller maintain the walk-count... (which it is... in my opinion...) we can globally change it at some point by finding all affected areas.

I know I don't need to convince you any more, just giving more clarity on my thoughts.